src/xdg-desktop-portal.c

```diff
@@ -196,26 +196,31 @@ on_bus_acquired (GDBusConnection *connection,
   PortalImplementation *implementation;
   PortalImplementation *lockdown_impl;
   PortalImplementation *access_impl;
-  g_autoptr(GError) error = NULL;
-  XdpDbusImplLockdown *lockdown;
+  XdpDbusImplLockdown *lockdown = NULL;
   GQuark portal_errors G_GNUC_UNUSED;
   GPtrArray *impls;
+  g_autoptr(GError) error = NULL;
 
   /* make sure errors are registered */
   portal_errors = XDG_DESKTOP_PORTAL_ERROR;
 
   xdp_connection_track_name_owners (connection, peer_died_cb);
-  init_document_proxy (connection);
-  init_permission_store (connection);
+
+  if (!init_permission_store (connection, &error))
+    g_error ("No permission store: %s", error->message);
```

Comment:
This will die with `SIGABRT` or equivalent, and dump core. Is that what we want?

If we want the portal to exit gracefully, the way to achieve that would be something like this:

```
static int global_exit_status = 0;
static GMainLoop *loop = NULL;

…

  if (!init_permission_store (connection, &error))
    {
      g_critical ("No permission store: %s", error->message);
      global_exit_status = 1;
      g_main_loop_quit (loop);
      return;
    }
  /* and the same for the documents proxy */

…

main (...)
{
  …

  loop = …;
  …
  g_main_loop_run (loop);
  …

  return global_exit_status;
}
```

Reply:
updated the PR to do this